### PR TITLE
♻️ improve cache keys

### LINKED ISSUE #18: Simplify board hashing
Currently, board is hashed via `json_encode`, that encodes the array of squares.
This produce a string that is hard to read and ridiculously long.
Then, such string is also used as a key, producing ugly arrays.

I think that we could use a FEN string, that is far shorter and simpler

### PR DESCRIPTION
**Pull request by garak**
This will fix #18

### CODE REVIEW

**Comment by garak** [comment]
@akondas ?

File: src/Board.php [hunk 1]
```diff
@@ -149,7 +149,7 @@ public function rewind(): ?Piece
 
     public function jsonSerialize(): string
     {
-        return \json_encode($this->squares);
+        return \implode(' ', $this->squares);
```
Comment (akondas) [inline src/Board.php:152]: you should probably change the name of the method, because it will do something else now

Comment (garak) [reply src/Board.php:152]: We can't change method name, it's needed to be able to call `json_encode()` on the object.

Moreover, it's perfectly fine to return a string in `jsonSerialize()`, that just need to return a "JSON representation of a value" (according to official php documentation)

Comment (akondas) [reply src/Board.php:152]: I am aware of this, but should this return string be a valid json string?

Comment (garak) [reply src/Board.php:152]: It's valid. This is an example:
`"r n b q k b n r p p p p p p p p                     P            P P P P  P P P R N B Q K B N R"`

I validated it on https://jsonlint.com/

Comment (akondas) [reply src/Board.php:152]: actually, just like the string itself, ok you convinced me :wink:

**Pull request merged**